online-store/src/components/search/search.ts

```diff
@@ -0,0 +1,102 @@
+import { productsArrayI } from '../types';
+import { productsData, CreateProductCard } from '../cards/cards';
+import { countView, resultData } from '../filter/filter';
+import { SortProductCard } from '../sort/sort';
+import { searchParams } from '../routing/routing';
+
+export let searchResult: Array<productsArrayI> = [];
+
+export function searchProductCard(
+  trigger: string,
+  arraySearch?: productsArrayI[]
+) {
+  let searchArrProducts: productsArrayI[] = productsData.products.slice();
+
+  function update() {
+    const search = document.querySelector('.search') as HTMLInputElement;
+    let changeInput = search.value;
+    const categoryContainer = document.querySelector(
+      '.category-container'
+    ) as HTMLElement; // блок слайдера stock
+    const brandContainer = document.querySelector(
+      '.brand-container'
+    ) as HTMLElement; // блок слайдера price
+    const stockContainer = document.querySelector(
+      '.stock-container'
+    ) as HTMLElement; // блок слайдера stock
+    const priceContainer = document.querySelector(
+      '.price-container'
+    ) as HTMLElement; // блок слайдера price
+
+    if (
+      categoryContainer.dataset.active === 'true' ||
+      brandContainer.dataset.active === 'true' ||
+      stockContainer.dataset.active === 'true' ||
+      priceContainer.dataset.active === 'true'
+    ) {
```

Comment:
there really is a need to use strings as the value boolean and not the boolean itself